torch/_inductor/fx_passes/comm_fusion.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import operator
+from typing import Any, Callable, Optional, Union
+
+import torch
+from torch._dynamo.utils import counters
+from torch._utils_internal import print_graph
+
+from ..pattern_matcher import (
+    CallFunctionVarArgs,
+    get_arg_value,
+    MULTIPLE,
+    PatternMatcherPass,
+    register_graph_pattern,
+)
+
+logger = logging.getLogger(__name__)
+
+
+comm_fusion_patterns = PatternMatcherPass()
+
+
+def comm_fusion_passes(graph: torch.fx.Graph) -> None:
+    """Pattern matching passes for collective communication fucntion fusion."""
+    if not torch.distributed.is_available:
```

Comment:
We should check the distributed availability in `post_grad.py` and assume distributed is available inside this module.

Reply:
Appreciate the suggestion, I think placing the distributed availability check within the communication fusion-only file helps keep a more self-contained structure. Given that the post_grad file is designed to be general, I prefer not to introduce this check there to avoid potential confusion in that broader context.